### PR TITLE
one fix per day #8

### PR DESCRIPTION
**Pull request by mbaechler**
…f network from DB

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java [hunk 1]
```diff
@@ -80,61 +78,40 @@ public void init() throws MessagingException {
...
-            return matched;
+        String recipientUser = recipientMailAddress.getLocalPart().toLowerCase(Locale.US);
+        String recipientHost = recipientMailAddress.getDomain().asString();
+        try (Connection conn = datasource.getConnection()) {
+            return Flux
+                .concat(
+                    Mono.fromCallable(() -> loadNetworks(recipientUser, recipientHost, conn)),
+                    Mono.fromCallable(() -> loadNetworksForHost(recipientHost, conn)))
+                .map(nets -> new NetMatcher(nets, dns))
+                .any(matcher -> matcher.matchInetNetwork(mail.getRemoteAddr()))
```
Comment (remk) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: you don't do those IO operations on a BoundedElastic Scheduler ?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java [hunk 2]
```diff
@@ -80,61 +78,40 @@ public void init() throws MessagingException {
...
-        } finally {
-            theJDBCUtil.closeJDBCConnection(conn);
         }
     }
 
+    private List<String> loadNetworksForHost(String recipientHost, Connection conn) throws SQLException {
+        return loadNetworks("*", recipientHost, conn);
+    }
+
+    private List<String> loadNetworks(String recipientUser, String recipientHost, Connection conn) throws SQLException {
```
Comment (remk) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: Could return a Mono ?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java (hunk 1 above)
Comment (mbaechler) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: mailets are synchronous so switching from "same thread" to another scheduler will just make things slower.

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java (hunk 2 above)
Comment (mbaechler) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: The only reason I used Flux over Stream is that it's easier to manage exceptions but using Mono here would let the caller think it's asynchronous when it's not

**Comment by chibenwa** [comment]
Was :green_apple:

File: server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java [hunk 3]
```diff
@@ -80,61 +78,40 @@ public void init() throws MessagingException {
...
+        List<String> nets = new ArrayList<>();
```
Comment (nvduc91) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: Maybe `LinkedList` is better for add/remove than `ArrayList`

Comment (mbaechler) [reply server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/NetworkIsInWhitelist.java]: There's no removal and appending is probably faster on an ArrayList.
BTW, I don't expect a lot of entries here so it should not make any difference in real life.

**Comment by Arsnael** [comment]
Was :green_apple: before CI restart

**Comment by mbaechler** [comment]
Don't loose time making this green, it requires to write some tests against a memory database and I don't have time right now to do it.

**Pull request closed without merging**